stage0/app.go

```diff
@@ -91,6 +100,12 @@ func AddApp(cfg AddConfig) error {
 		}
 	}
 
+	l, err := lock.ExclusiveLock(common.PodManifestLockPath(cfg.PodPath), lock.RegFile)
```

Comment:
Can you add a debugging message here? Since this could block, someday it *will* block and we'll want to be able to easily find why.